File: docs-examples/example-java/src/test/java/io/micronaut/configuration/openapi/docs/OpenApiSpec.java
package io.micronaut.configuration.openapi.docs;

import io.micronaut.openapi.OpenApiUtils;
import io.swagger.v3.oas.models.OpenAPI;

import java.io.IOException;
import java.io.InputStream;
import java.io.UncheckedIOException;

/**
 * Loads the OpenAPI document generated at compile time into the class output.
 */
final class OpenApiSpec {

    private OpenApiSpec() {
    }

    static OpenAPI load() {
        return load("hello-world-0.0.yml");
    }

    static OpenAPI load(String fileName) {
        try (InputStream stream = OpenApiSpec.class.getResourceAsStream("/META-INF/swagger/" + fileName)) {
            if (stream == null) {
                throw new IllegalStateException("Generated OpenAPI document not found: " + fileName);
            }
            return OpenApiUtils.getYamlMapper().readValue(stream, OpenAPI.class);
        } catch (IOException e) {
            throw new UncheckedIOException(e);
        }
    }
}

File: docs-examples/example-groups-kotlin/src/test/kotlin/io/micronaut/configuration/openapi/docs/groups/OpenApiSpec.kt
package io.micronaut.configuration.openapi.docs.groups

import io.micronaut.openapi.OpenApiUtils
import io.swagger.v3.oas.models.OpenAPI

/**
 * Loads the OpenAPI document generated at compile time into the class output.
 */
object OpenApiSpec {

    fun load(fileName: String): OpenAPI {
        val stream = OpenApiSpec::class.java.getResourceAsStream("/META-INF/swagger/$fileName")
            ?: throw IllegalStateException("Generated OpenAPI document not found: $fileName")
        return stream.use { OpenApiUtils.getYamlMapper().readValue(it, OpenAPI::class.java) }
    }
}
